go/app/main.go

```diff
@@ -39,117 +41,206 @@ type Items struct {
 	Items []Item `json:"items"`
 }
 
-func addItem(c echo.Context) error {
-	// リクエストボディからデータを取得
-	name := c.FormValue("name")
-	category := c.FormValue("category")
+type ServerImpl struct {
+	db *sql.DB
+}
 
-	// 画像ファイルを取得
-	imageFile, err := c.FormFile("image")
-	if err != nil {
-		res := Response{Message: "Failed to get image file"}
-		return c.JSON(http.StatusBadRequest, res)
-	}
-	src, err := imageFile.Open()
-	if err != nil {
-		res := Response{Message: "Failed to open image file"}
-		return c.JSON(http.StatusInternalServerError, res)
-	}
-	defer src.Close()
+func (s ServerImpl) addItem() echo.HandlerFunc {
```

Comment:
nits: 以下の実装をすれば元のインタフェースを保持したまま関数からメソッドへ置換できます :+1:

```suggestion
func (s ServerImpl) addItem(c echo.Context) error {
  // snip
}

func main() {
  // snip
  e.POST("/items", serverImpl.addItem)
}
```

Reply:
なるほど、ありがとうございます！✨
[562a44e](https://github.com/dongurikoko/mercari-build-training/pull/3/commits/562a44e39159a4856c03cfdc7bab0602bfd691f9)にて修正しました！